llvm/lib/Transforms/Utils/SCCPSolver.cpp

```diff
@@ -1627,7 +1619,10 @@ void SCCPInstVisitor::visitSelectInst(SelectInst &I) {
   if (ConstantInt *CondCB =
           getConstantInt(CondValue, I.getCondition()->getType())) {
     Value *OpVal = CondCB->isZero() ? I.getFalseValue() : I.getTrueValue();
-    mergeInValue(&I, getValueState(OpVal));
+    const ValueLatticeElement &OpValState = getValueState(OpVal);
+    // Safety: ValueState[&I] doesn't invalidate OpValState since it is already
+    // in the map.
+    mergeInValue(ValueState[&I], &I, OpValState);
```

Comment:
It's unfortunate that `ValueState.at(&I)` returns a const reference, otherwise we could use that to assert existence.

Reply:
I added an assertion here.